src/server_name.rs

```diff
@@ -49,6 +51,18 @@ impl<'a> ServerName<'a> {
             Self::IpAddress(i) => ServerName::IpAddress(*i),
         }
     }
+
+    /// Return the string representation of this `ServerName`.
+    #[cfg(all(feature = "alloc", feature = "std"))]
+    pub fn to_str(&self) -> Cow<'_, str> {
+        match self {
+            Self::DnsName(d) => d.as_ref().into(),
+            Self::IpAddress(i) => {
+                let ip: std::net::IpAddr = (*i).into();
+                ip.to_string().into()
+            }
```

Comment:
I think we could make this not require `std` if we wanted to lift the `fmt::Display` impl from core for both the `Ipv4Addr` and `Ipv6Addr` types in this crate. I wasn't sure if it was worth the effort in this case. WDYT?

Reply:
Probably not worth the effort, the IPv6 one is kinda gnarly.